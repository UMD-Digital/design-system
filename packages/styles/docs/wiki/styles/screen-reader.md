[umd-styles-library](../wiki/Home) / screen-reader

# screen-reader

## Variables

| Variable | Description |
| ------ | ------ |
| [only](../wiki/screen-reader.Variable.only) | Screen reader only class. |
