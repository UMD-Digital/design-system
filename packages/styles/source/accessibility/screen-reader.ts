import { create } from '../utilities';
import type { JssObject } from '../_types';

/**
 * Screen reader only class.
 * @returns {JssObject} The JSS object for the screen reader only class.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.accessibility.screenReader.only
 * ```
 * @example
 * ```css
 * class="sr-only"
 * ```
 * @since 1.1.0
 */
export const only: JssObject = create.jss.objectWithClassName({
  className: 'sr-only',
  clip: 'rect(0,0,0,0)',
  borderWidth: '0px',
  height: '1px',
  margin: '-1px',
  overflow: 'hidden',
  padding: '0',
  position: 'absolute',
  whiteSpace: 'nowrap',
  width: '1px',
});
