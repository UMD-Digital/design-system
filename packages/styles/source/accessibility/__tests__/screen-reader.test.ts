import { only } from '../screen-reader';

describe('accessibility/screen-reader', () => {
  describe('only', () => {
    it('should match snapshot', () => {
      expect(only).toMatchSnapshot();
    });

    it('should have correct className', () => {
      expect(only.className).toBe('sr-only');
    });

    it('should have appropriate styles for screen reader only content', () => {
      expect(only.clip).toBe('rect(0,0,0,0)');
      expect(only.position).toBe('absolute');
      expect(only.width).toBe('1px');
      expect(only.height).toBe('1px');
      expect(only.overflow).toBe('hidden');
      expect(only.whiteSpace).toBe('nowrap');
    });
  });
});