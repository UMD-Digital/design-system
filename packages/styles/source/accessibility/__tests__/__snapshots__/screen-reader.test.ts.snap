// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accessibility/screen-reader only should match snapshot 1`] = `
{
  "borderWidth": "0px",
  "className": "sr-only",
  "clip": "rect(0,0,0,0)",
  "height": "1px",
  "margin": "-1px",
  "overflow": "hidden",
  "padding": "0",
  "position": "absolute",
  "whiteSpace": "nowrap",
  "width": "1px",
}
`;
